File: nodes/delete-object-property/delete-object-property.process.js
/**
 * Process function for the Delete Object Property node.
 * Remove a specific property from an object, returning a new object without that property.
 */
export default async ({inputs, settings, config}) => {
  const { object, key } = inputs;
  
  // Validate inputs
  if (!object || typeof object !== 'object') {
    return {
      object: {},
      key: key || '',
      removed_value: null,
      was_removed: false
    };
  }
  
  if (!key || typeof key !== 'string') {
    return {
      object: object,
      key: key || '',
      removed_value: null,
      was_removed: false
    };
  }
  
  // Check if property exists
  const was_removed = key in object;
  const removed_value = was_removed ? object[key] : null;
  
  // Create new object without the property
  const newObject = { ...object };
  delete newObject[key];
  
  return {
    object: newObject,
    key: key,
    removed_value: removed_value,
    was_removed: was_removed
  };
};

File: nodes/delete-object-property/delete-object-property.process.py
"""
Process function for the Delete Object Property node.
Remove a specific property from an object, returning a new object without that property.
"""
from typing import Any, Dict

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    object_input = inputs.get("object")
    key = inputs.get("key")
    
    # Validate inputs
    if not isinstance(object_input, dict):
        return {
            "object": {},
            "key": key or "",
            "removed_value": None,
            "was_removed": False
        }
    
    if not isinstance(key, str):
        return {
            "object": object_input,
            "key": key or "",
            "removed_value": None,
            "was_removed": False
        }
    
    # Check if property exists
    was_removed = key in object_input
    removed_value = object_input.get(key) if was_removed else None
    
    # Create new object without the property
    new_object = object_input.copy()
    if was_removed:
        del new_object[key]
    
    return {
        "object": new_object,
        "key": key,
        "removed_value": removed_value,
        "was_removed": was_removed
    }

